### ISSUE TITLE
Generating API access token throws error:0909006C:PEM

### ISSUE THREAD

**Issue by R0Wi** [salesagility/SuiteCRM-Core#185]
When trying to create a new API access token, I get the following error message:

```
It was not possible to parse your key, reason: error:0909006C:PEM routines:get_name:no start line
```

Preparations for API usage have been done like stated in

* https://docs.suitecrm.com/developer/api/developer-setup-guide/json-api/#_before_you_start_calling_endpoints and
* https://docs.suitecrm.com/developer/api/developer-setup-guide/configure-authentication/

#### Issue
When calling the API access token endpoint `{{server}}/legacy/Api/access_token` via POST request, the server gives me an error 500 with the following JSON response:

```json
{
    "error": "unknown_error",
    "message": "It was not possible to parse your key, reason: error:0909006C:PEM routines:get_name:no start line"
}
```

#### Expected Behavior

Server response 200 containing a valid access token

#### Actual Behavior

Response mentioned above. No furhter logs.

#### Possible Fix

The projects uses `"lcobucci/jwt": "3.3.3"` 

https://github.com/salesagility/SuiteCRM-Core/blob/91064df88284dc3cd6a3e5a268185a1025f8bd0f/composer.json#L49

which officially doesn't support PHP 8.0 (see https://packagist.org/packages/lcobucci/jwt#3.3.3). Updating this dependency might help but I'm not sure if this is the root cause. Like stated in some of the source below, this error seems to be related to the PHP version. Some mention that a downgrade to PHP 7.4 would help. Since this version is already [deprecated](https://www.php.net/supported-versions.php), that's no option for me.

#### Steps to Reproduce

1. Setup SuiteCRM for API usage
2. Call `{{server}}/legacy/Api/access_token` with valid credentials

#### Your Environment

* SuiteCRM Version used: 8.2.1
* Environment name and version: MySQL, PHP 8.0
* Operating System and version: Docker, docker-compose with [Bitnami image](https://hub.docker.com/r/bitnami/suitecrm/)

#### Links to similar issues

* Bitnami Image issue: https://github.com/bitnami/containers/issues/17280
* Forum: https://community.suitecrm.com/t/api-and-docker-error-it-was-not-possible-to-parse-your-key/84638/2
* https://github.com/lcobucci/jwt/issues/310
* https://stackoverflow.com/questions/67733476/when-i-login-then-error-will-be-ocuurred-it-was-not-possible-to-parse-your-key
* https://stackoverflow.com/questions/73211594/lombucci-jwt-it-was-not-possible-to-parse-your-key-reason-error0909006cpem-r

**Comment by R0Wi** [comment]
I did some more research and debugging and here are my results:

indeed the error seems to come from `lcobucci/jwt` library. PHP 8.0 compatibility was introduced by https://github.com/lcobucci/jwt/commit/be0f2f94c4bd24a2c69a6af9e5949848c29a0921#diff-6019fde4d9e0d7d2d060b7fd9d8ebd89f85be6d89f61de292660c9a01bea8ca9R84 which fixed a breaking change in PHP 8.0. According to the [docs](https://www.php.net/manual/en/function.openssl-pkey-get-private.php) for `openssl_pkey_get_private`:

> 8.0.0 | On success, this function returns an OpenSSLAsymmetricKey instance now; previously, a resource of type OpenSSL key was returned

I applied the patch of that one line manually and was able to create access tokens again. So an update of `lcobucci/jwt` to >= 4.0.0 would be sufficient. But since this is also a dependency of `league/oauth2-server`, that package would also need to be upgraded. This in turn introduces some breaking code API changes which have to be implemented first. 

If upgrading the packages accordingly and adjusting the code to the new API would be a solution for you, I could image creating a PR for this.

**Comment by corearchitect** [comment]
> I applied the patch of that one line manually and was able to create access tokens again.

Which specific line did you patch manually?

**Comment by R0Wi** [comment]
[This line here](https://github.com/lcobucci/jwt/commit/be0f2f94c4bd24a2c69a6af9e5949848c29a0921#diff-6019fde4d9e0d7d2d060b7fd9d8ebd89f85be6d89f61de292660c9a01bea8ca9R84). Basically just replacing `! is_resource($key)` with `is_bool($key)` in the `jwt` vendors subfolder which is created by composer.

If you scroll up, you can see that the patch is included in `jwt >= 4.0.0`
![image](https://user-images.githubusercontent.com/19730957/227951185-aea63224-2b9e-40ad-8789-ad8d52366241.png)


So I guess upgrading this dependency should fix the problem.

**Comment by florian-besser** [comment]
Hey @johnM2401 I think https://github.com/salesagility/SuiteCRM/pull/9965 did indeed fix the issue. We're using the `bitnami` helm charts to install SuiteCRM and using the newest SuiteCRM version `7.13.4` (from docker image `docker.io/bitnami/suitecrm:7.13.4-debian-11-r15`) does indeed no longer need these fixes. 

As long as `/bitnami/suitecrm/Api/V8/OAuth2/private.key` and `/bitnami/suitecrm/Api/V8/OAuth2/public.key` are present it'll read them without issues.
